src/libraries/System.Text.Json/System.Text.Json.SourceGeneration.UnitTests/JsonSourceGeneratorTests.cs

```diff
@@ -2,32 +2,222 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Immutable;
+using System.Text.Json.Serialization;
+using System.Diagnostics.CodeAnalysis;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Xunit;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Linq;
+using System.IO;
 
 namespace System.Text.Json.SourceGeneration.UnitTests
 {
-    public static class GeneratorTests
+    public class GeneratorTests
     {
         [Fact]
-        public static void SourceGeneratorInitializationPass()
+        public void TypeDiscoveryPrimitivePOCO()
```

Comment:
It seems that there is a lot of common code between this test method and the one below. Can we extract some of the logic into a helper class/helper methods?